### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by AbrahaKahsay**
# Project description

Now it's time to set up our models. To start with, we have tables for Users, Posts, Comments, and Likes, which means we need a model for each one. You've already set the foreign key in the table schema but be sure to write it here too!

## Project requirements
**Create models**
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.

**Use models to insert data**
- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:
   - first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
   -  second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
- Create at least 4 posts written by one of the users you created by running the following code:
   - first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
- Create at least 6 posts comments for one of the posts you created by running the following code:
   - Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

**Create custom methods**

Your models should include some additional methods.
**Users**
- A method that returns the 3 most recent posts for a given user.

**Posts**
- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.

**Comments**
- A method that updates the comments counter for a post.

**Likes**
- A method that updates the likes counter for a post.
- Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :likes
+
+  def update_posts_counter
+    user.increment!(:posts_counter)
+  end
+
+  def recent_five_comments
+    comments.order!(created_at: :desc).limit(5)
+  end
+end
```
Comment (GzimAsani) [inline app/models/post.rb]: - [ ] You have a problem with your migration, kindly take a step back and update it
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello    @AbrahaKahsay       :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * The necessary models  are created :+1: 

 ## Required Changes 

- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (AbrahaKahsay) [reply app/models/post.rb]: Hi, @GzimAsani thanks for the review. great catch. I tried to re-generate the migrate files and it worked fine but In the console to create posts, it is giving me an error as in the figure. I couldn't figure it out, could you help me?
Screenshot from 2022-10-20 21-24-53

**Review by alexander16108 — APPROVED** [review]
Hi @AbrahaKahsay , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**